### PR TITLE
fix(http): http min works even when the main http server is disabled

### PR DESCRIPTION
**Pull request by jerrinot**
This changeset also separates `getMinHttpSocketFactory()` from `getHttpSocketFactory()` so the min HTTP server can use a different socket factory than the main HTTP server.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 67 / 68 (98.53%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|62|63|98.41%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 67 (98.51%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|59|60|98.33%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|

File: core/src/main/java/io/questdb/PropServerConfiguration.java [hunk 1]
```diff
@@ -607,99 +607,100 @@ public PropServerConfiguration(
...
-                final String publicDirectory = getString(properties, env, PropertyKey.HTTP_STATIC_PUBLIC_DIRECTORY, "public");
-                // translate public directory into absolute path
-                // this will generate some garbage, but this is ok - we're just doing this once on startup
-                if (new File(publicDirectory).isAbsolute()) {
-                    this.publicDirectory = publicDirectory;
-                } else {
-                    this.publicDirectory = new File(root, publicDirectory).getAbsolutePath();
-                }
+            this.connectionStringPoolCapacity = getInt(properties, env, PropertyKey.HTTP_CONNECTION_STRING_POOL_CAPACITY, 128);
+//            if (httpServerEnabled) {
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/PropServerConfiguration.java:615]: Looks like a leftover.

File: core/src/test/java/io/questdb/test/cutlass/http/HttpMinBootstrapTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Before
+    public void setUp() {
+        super.setUp();
+        TestUtils.unchecked(() -> createDummyConfiguration());
+        dbPath.parent().$();
+    }
+
+    @Test
+    public void testHttpServerEnabledMinStillWorking() throws Exception {
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/cutlass/http/HttpMinBootstrapTest.java:35]: nit: the test name should be `testHttpServerDisabledMinServerStillWorking`

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 67 (98.51%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|59|60|98.33%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|

**Review by puzpuzpuz — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
Nice catch!

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 67 (98.51%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|59|60|98.33%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/Services.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cutlass/Services.java [hunk 3]
```diff
@@ -87,7 +87,7 @@ public static HttpServer createHttpServer(
             return null;
         }
 
-        final HttpServer server = new HttpServer(configuration, cairoEngine.getMessageBus(), metrics, workerPool);
+        final HttpServer server = new HttpServer(configuration, cairoEngine.getMessageBus(), metrics, workerPool, configuration.getFactoryProvider().getHttpSocketFactory());
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cutlass/Services.java:90]: this is formatting i am constantly undoing, would you mind changing?

it is either this, if call fits a line:

```java
void func(int a, int b, int c);
```
or 

```java
void func(
   int a,
   int b,
   int c
)
```

Not good:

```java
void func(int a, int b
   int c, int d);
```

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/cutlass/Services.java (hunk 3 above)
Comment (bluestreak01) [reply core/src/main/java/io/questdb/cutlass/Services.java:90]: ooops, GH rendering issue, code is fine

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**